frontend/ui/widget/touchmenu.lua

```diff
@@ -744,6 +746,105 @@ function TouchMenu:updateItems()
     end)
 end
 
+local MAX_MENU_DEPTH = 20 -- currently our menu needs at least 12 here
+function TouchMenu:_recurse(val, path, text, search_for, depth)
+    depth = depth + 1
+    if depth > MAX_MENU_DEPTH then
+        return
+    end
+    for i,v in pairs(val) do
+        local menu_text = nil
+
+        if type(v) == "table" and (type(i) == "number" or i:find("item")) then
+            local menu_entry = val[i].text or (val[i].text_func and  val[i].text_func())
+            local next_text = menu_entry and (text .. "→ " .. tostring(menu_entry)) or text -- drop sub_item_table
+            TouchMenu:_recurse(val[i], path .. "." .. i, next_text, search_for, depth)
+        elseif i == "text_func" then
+            menu_text = v()
+        elseif i == "text" then
+            menu_text = v
+        end
+
+        if menu_text and Utf8Proc.lowercase(menu_text):find(search_for) then
+            table.insert(TouchMenu.foundMenuItem, {menu_text, path .. "." .. i, text})
+        end
+    end
+end
+
+function TouchMenu:onMenuSearch(search_for)
+    TouchMenu.foundMenuItem = {}
+    for i = 1, #self.tab_item_table do
+        TouchMenu:_recurse(self.tab_item_table[i], i, self.tab_item_table[i].text or "ROOT"..i, search_for, 0)
+    end
+
+--[[
+    for i = 1, #TouchMenu.foundMenuItem do
+        print("xxxxxxx->", i,
+            TouchMenu.foundMenuItem[i][1],
+            TouchMenu.foundMenuItem[i][2],
+            TouchMenu.foundMenuItem[i][3])
+    end
+]]
+end
+
+function TouchMenu:onOpenMenu(path, animate_time_s)
+    local function animate_func()
+        if not animate_time_s or animate_time_s == 0.0 then return end
+        local sleep_us = animate_time_s * 1e6
+        UIManager:forceRePaint()
+        ffiUtil.usleep(sleep_us)
+        UIManager:setDirty(nil, "ui")
+    end
+    -- first switch to correct MenuTab
+    local sep_pos = path:find("%.")
+    local tab_num = tonumber(path:sub(1, sep_pos - 1))
+    path = path:sub(sep_pos + 1)
+    local item = self.tab_item_table[tab_num]
+
+    if not tab_num then
+        return
+    end
+    self:switchMenuTab(tab_num)
+    self.bar:switchToTab(tab_num)
+
+    self:updateItems()
+
+    -- Now open the menus all the way down
+    local dummy, num_of_sep = path:gsub("%.", "%.")
+    while num_of_sep > 1 do
+        sep_pos = path:find("%.")
+        local identifier = path:sub(1, sep_pos -1)
+        local item_num = tonumber(identifier)
+        path = path:sub(sep_pos + 1)
+        if item_num then
+            animate_func()
+            item = item[item_num]
+            self:onMenuSelect(item)
```

Comment:
The animation is fun to watch and wait (but mostly to code :) - but it lacks indicating which menu items needs to be tapped when navigating down - which makes it a bit useless as a learning tool.
It could either be the "Flash buttons and menu items" menu item transitional inversion (without the final in-inversion being needed), or some finger/large dot on the right to indicate which menu to go into. I think the former would be more consistent (and should be enabled also for people like me who have "[ ] Flash buttons" disabled, who don't need it when it's my actual finger doing the action).

Reply:
Sounds sensible (and more readable/in line with our usual stuff than a "cursor" like Android screen recording does would) ;).

IIRC, `widgetInvert` should work on Menu labels.